pkg/plugin/dns/dns_linux.go

```diff
@@ -17,21 +17,25 @@ import (
 	"github.com/microsoft/retina/pkg/enricher"
 	"github.com/microsoft/retina/pkg/log"
 	"github.com/microsoft/retina/pkg/metrics"
-	"github.com/microsoft/retina/pkg/plugin/api"
 	"github.com/microsoft/retina/pkg/plugin/common"
+	"github.com/microsoft/retina/pkg/plugin/registry"
 	"github.com/microsoft/retina/pkg/utils"
 	"go.uber.org/zap"
 )
 
-func New(cfg *kcfg.Config) api.Plugin {
+func init() {
+	registry.Plugins[name] = New
```

Comment:
I might make this a method of the registry, just so that you can loudly panic if there's a name collision. Unlikely to be an issue in the near term, but you never know.

Reply:
Good idea, don't want to let a new plugin clobber an existing registration.
I had started to do it [here](https://github.com/microsoft/retina/blob/7e2d7130c03779a3334f6b693a37e0d2205b3a3e/pkg/plugin/registry/registry.go#L42-L44), let me finish hiding the Registry from direct txs

Reply:
updated